http.go

```diff
@@ -92,7 +103,7 @@ func (client *Client) Request(rawPayload []byte, method, route string, query url
 				return nil, nil, err
 			}
 		}
-		resp, err := (&http.Client{}).Do(callSummary.HTTPRequest)
+		resp, err := DefaultHTTPClient.Do(callSummary.HTTPRequest)
```

Comment:
I am not sure about removing the interface. The fact that it has only one method now doesn't mean it won't have more in the future. Bear in mind that I am talking about general software design, if another method inside the interface doesn't make sense by design, go ahead with replacing the interface by a function.